src/oom/oomd-manager.c

```diff
@@ -652,6 +696,37 @@ static int manager_connect_bus(Manager *m) {
         return 0;
 }
 
+static int manager_varlink_init(Manager *m) {
+        _cleanup_(varlink_server_unrefp) VarlinkServer *s = NULL;
+        int r;
+
+        assert(m);
+        assert(!m->varlink_server);
+
+        r = varlink_server_new(&s, VARLINK_SERVER_ACCOUNT_UID|VARLINK_SERVER_INHERIT_USERDATA);
+        if (r < 0)
+                return log_error_errno(r, "Failed to allocate varlink server object: %m");
+
+        varlink_server_set_userdata(s, m);
+
+        r = varlink_server_bind_method(s, "io.systemd.oom.SendManagedOOMCGroups", process_managed_oom_request);
+        if (r < 0)
+                return log_error_errno(r, "Failed to register varlink method: %m");
+
+        r = varlink_server_listen_fd(s, SD_LISTEN_FDS_START + 0);
```

Comment:
i'd prefer if we could run this from the cmdline for debug purposes, i.e. in gdb and this would still work. i.e. use the listen fd logic if there is a socket passed, otherwise just bind the socket yourself